File: Q1/vernam_cipher.py
def vernam_chiper(key, message):
    xor = []

    for i in range(len(key)):
        c_xor = ord(key[i]) ^ ord(message[i])
        xor.append(chr(c_xor))

    return ''.join(xor)

def main():
    M = "HELLO"
    K = "pizza"
    
    C = vernam_chiper(M, K)
    print(C)

main()

File: chiff/Q1/vernam_cipher.cc
#include <iostream>

std::string vernam_cipher(std::string key, std::string message)
{
    int length = key.length();
    
    std::string result = "";

    for (int i = 0; i < length; ++i)
    {
        result += key[i] ^ message[i];
    }

    return result;
}

int main()
{
    std::string M = "HELLO";
    std::string K = "pizza";

    std::string C = vernam_cipher(M, K);

    std::cout << C << std::endl;

    return 0;
}